dss/operations/secrets.py

```diff
@@ -0,0 +1,312 @@
+def list_secrets(argv: typing.List[str], args: argparse.Namespace):
+    """
+    Print a list of names of every secret variable in the secrets manager
+    for the given store and stage
+    """
+    store_name, stage_name = get_secretsmanager_names(args)
+    arn_prefix, store_prefix = get_secretsmanager_prefixes(args)
+
+    paginator = secretsmanager.get_paginator('list_secrets')
+
+    secret_names = []
+    for response in paginator.paginate():
+        for secret in response['SecretList']:
+
+            # Get resource IDs
+            secret_name = secret['ARN']
+            short_secret_name, long_secret_name = long_short_resource_names(secret_name, arn_prefix, store_prefix)
+
+            # Only save secrets for this store and stage
+            if short_secret_name.startswith(store_prefix):
+                if args.long:
+                    secret_names.append(long_secret_name)
+                else:
+                    secret_names.append(short_secret_name)
+
+    secret_names.sort()
+
+    if args.json:
+        print(json.dumps(secret_names))
+    else:
+        for secret_name in secret_names:
+            print(secret_name)
+
+
+@events.action("get",
+               arguments={
+                   "--stage": dict(
+                       required=False,
+                       help="the stage for which secrets should be listed"),
+                   "--secret-name": dict(
+                       required=True,
+                       nargs="*",
+                       help="name of secret or secrets to retrieve (list values can be separated by a space)"),
+                   "--long": dict(
+                       default=False,
+                       action="store_true",
+                       help="use long (full ARN) identifiers when printing secret variable(s)"),
+                   "--json": dict(
+                       default=False,
+                       action="store_true",
+                       help="format the output as JSON if this flag is present")})
+def get_secret(argv: typing.List[str], args: argparse.Namespace):
+    """
+    Get the value of the secret variable (or list of variables) specified by
+    the --secret-name flag
+    """
+    # Note: this function should not print anything except the final JSON,
+    # in case the user pipes the JSON output of this script to something else
+    store_name, stage_name = get_secretsmanager_names(args)
+    arn_prefix, store_prefix = get_secretsmanager_prefixes(args)
+
+    # Make sure a secret name was specified
+    if args.secret_name is None or len(args.secret_name) == 0:
+        logger.error("Unable to set secret: no secret name was specified! Use the --secret-name flag.")
+        sys.exit()
+    secret_names = args.secret_name
+
+    for secret_name in secret_names:
+
+        # Get resource IDs
+        short_secret_name, long_secret_name = long_short_resource_names(secret_name, arn_prefix, store_prefix)
```

Comment:
I'm not sure I understand why the ARN is being constructed. It is not required to fetch the secret:
```
In [6]: import boto3                                                                                                                
In [7]: s = boto3.client("secretsmanager").get_secret_value(SecretId="dcp/dss/dev/es_source_ip")
```

The boto3 documentation also indicates this:
> You can specify either the Amazon Resource Name (ARN) or the friendly name of the secret.

Reply:
@xbrianh to provide full context, I know now that you can provide  a short name for a variable. However, AWS adds an alphanumeric suffix to each variable short name, and when you include the alphanumeric suffix with the variable short name, it can't find the variable.

For example, if you print all secrets, the `es_source_ip` variable is actually called

```
dcp/dss/dev/es_source_ip-q2baD1
```

and if you try and look up the variable using that exact short name, including the suffix, it will not find the variable.

See https://github.com/HumanCellAtlas/data-store/pull/2331 for more details - I hard-coded some calls to `get_secret_value()` with different forms of secret names to demonstrate what worked and what didn't. I discovered the suffix thing after I created that pull request.

Reply:
The suffix is included in the ARN, not the secret name. Check out this response to `list_secrets`:
```
{'ARN': 'arn:aws:secretsmanager:us-east-1:861229788715:secret:dcp/dss/dev/es_source_ip-q2baD1',
 'Name': 'dcp/dss/dev/es_source_ip',
 'LastChangedDate': datetime.datetime(2019, 8, 8, 19, 12, 6, 153000, tzinfo=tzlocal()),
 'LastAccessedDate': datetime.datetime(2019, 8, 8, 0, 0, tzinfo=tzlocal()),
 'SecretVersionsToStages': {'658c3b41-0806-48b9-b05d-ea7dc2dbf237': ['AWSCURRENT'],
  'f37ccfe2-16e0-4305-a250-ef89d2c47ece': ['AWSPREVIOUS']}}
```

Reply:
I have removed the functions and infrastructure for handling either long or short secret names separately, and instead converted everything in the script to use short names _only_ for retrieving secrets.